frontend/src/static/js/v_summary.js

```diff
@@ -362,33 +361,88 @@ window.vSummary = {
       return null;
     },
     sortFiltered() {
-      const full = [];
-      if (!this.filterGroupRepos) {
-        full.push([]);
-      }
-
-      this.filtered.forEach((users) => {
-        if (this.filterGroupRepos) {
-          users.sort(comparator((ele) => ele[this.filterSort]));
-          full.push(users);
-        } else {
-          users.forEach((user) => full[0].push(user));
-        }
-      });
-
-      if (!this.filterGroupRepos) {
-        full[0].sort(comparator((ele) => {
-          const field = ele[this.filterSort];
-          return field.toLowerCase ? field.toLowerCase() : field;
-        }));
+      let full = [];
+      if (this.filterGroupSelection === 'groupByNone') {
+        full[0] = this.filterByNone(this.filtered);
+      } else if (this.filterGroupSelection === 'groupByAuthors') {
+        full = this.filterByAuthors(this.filtered);
+      } else {
+        full = this.filterByRepos(this.filtered);
       }
 
       if (this.filterSortReverse) {
         full.forEach((repo) => repo.reverse());
+        if (this.filterGroupSelection === 'groupByAuthors') {
+          full.reverse();
+        }
       }
 
       this.filtered = full;
     },
+
+    filterByRepos(repos) {
+      const sortedRepos = [];
+      repos.forEach((users) => {
+        users.sort(comparator((ele) => ele[this.filterSort]));
+        sortedRepos.push(users);
+      });
+      return sortedRepos;
+    },
+    filterByNone(repos) {
+      const sortedRepos = [];
+      repos.forEach((users) => {
+        users.forEach((user) => {
+          sortedRepos.push(user);
+        });
+      });
+      sortedRepos.sort(comparator((ele) => {
+        const field = ele[this.filterSort];
+        return field.toLowerCase ? field.toLowerCase() : field;
+      }));
+      return sortedRepos;
+    },
+    filterByAuthors(repos) {
+      const authorMap = {};
+      const authorData = {};
```

Comment:
if you never use `authorData` until later, put it tgt below, where you actually use it. makes code easier to trace